cli/cmd/upgrade.go

```diff
@@ -252,6 +257,15 @@ func upgrade(ctx context.Context, k *k8s.KubernetesAPI, flags []flag.Flag, stage
 		}
 	}
 
+	// If values is still nil, then neither the linkerd-config-overrides secret
+	// nor the legacy values were found. This means either means that Linkerd
+	// was installed with Helm or that the installation needs to be repaired.
+	if values == nil {
+		return bytes.Buffer{}, errors.New("Could not find the Linkerd config. If Linkerd was installed with Helm, " +
+			"please use Helm to perform upgrades. If Linkerd was not installed with Helm, please use the " +
+			"`linkerd repair` command to repair the Linkerd config")
```

Comment:
nit: Maybe use \` in a `fmt.Sprintf` here? It should pick up the newlines and avoid the use of `+`.

Reply:
I'm not sure what you mean.  I've used `+` concatenation to split this across multiple lines without introducing any newlines into the string itself.  I assume the go compiler is smart enough to turn this all into a single statically allocated string.

Reply:
Yep never mind my mistake.